### PR TITLE
Comments: Add stats to Comment Management actions

### LINKED ISSUE #16644: Comments: add stats to actions
- make sure page view stats are firing correctly
- tracks/bump stats for all comment actions

Double check with post analytics for a similar naming style.

### PR DESCRIPTION
**Pull request by Copons**
Closes #16644 

Add stats tracking to all the Comment Management actions.

All Tracks event are namespaced as `calypso_comment_management_`.
All stats bump are on the `calypso_comment_management` group.

### Events

| Action | Bump | Tracks | Additional Info |
| --- | --- | --- | --- |
| Change Status | `comment_status_changed_to_{status}` | `change_status` | `alsoUnlike`<br>`isUndo`<br>`status` |
| Delete | `comment_deleted` | `delete` ||
| Like | `comment_liked` | `like` | `alsoApprove` |
| Change Page | `change_page` | `change_page` | `page`<br>`total` |
| Reply | `comment_reply` | `reply` | `alsoApprove` |
| Unlike | `comment_unliked` | `unlike` ||
| Change Filter | `change_filter` | `change_filter` | `status` |

### Additional Info

- Change Status
  - `alsoUnlike`: when a liked comment is unapproved, marked as spam, or moved to trash, it also loses its like.
  - `isUndo`: marks if the Change Status action was an undo.
  - `status`: the new comment status.
- Like
  - `alsoApprove`: when an unapproved comment is liked, it also becomes approved.
- Change Page
  - `page`: the page requested.
  - `total`: the total count of pages at the moment of request.
- Reply
  - `alsoApprove`: when replying to an unapproved, spam, or trash comment, that comment becomes approved.
- Change Filter
  - `status`: the new Comment List filter.

### CODE REVIEW

File: client/my-sites/comments/comment-navigation/index.jsx [hunk 1]
```diff
@@ -194,4 +204,11 @@ export class CommentNavigation extends Component {
 	}
 }
 
-export default localize( UrlSearch( CommentNavigation ) );
+const mapDispatchToProps = {
+	recordChangeFilter: status => composeAnalytics(
+		recordTracksEvent( COMMENTS_STATS_GROUP + '_change_filter', { status } ),
```
Comment (gwwar) [inline client/my-sites/comments/comment-navigation/index.jsx]: Not blocking but I usually like these as strings, so they're easy to search for. So for example:
```jsx
recordTracksEvent( 'calypso_comment_management_change_filter', { status } )
```

Comment (Copons) [inline client/my-sites/comments/comment-navigation/index.jsx]: Got it, and it makes perfect sense.

Comment (gwwar) [inline client/my-sites/comments/comment-navigation/index.jsx]: I think we can avoid creating a new function on each render, how about:
```jsx
const mapDispatchToProps = function( dispatch, { status } ) {
    return {
        recordChangeFilter: () => { 
            dispatch( composeAnalytics(
                   recordTracksEvent( 'calypso_comment_management_change_filter', { status } ),
                   bumpStat( calypso_comment_management, `change_filter_to_${ status }` )
            ) );
    };
};
```

File: client/my-sites/comments/comment-list/index.jsx [hunk 2]
```diff
@@ -93,10 +102,15 @@ export class CommentList extends Component {
 		}
 	}
 
-	changePage = page => this.setState( {
-		page,
-		selectedComments: {},
-	} );
+	changePage = page => {
+		const total = Math.ceil( ( this.props.comments.length + this.state.persistedComments.length ) / COMMENTS_PER_PAGE );
```
Comment (gwwar) [inline client/my-sites/comments/comment-list/index.jsx]: Not sure we need the total here. If we're interested in that, maybe we could add it at the pager level instead.

File: client/my-sites/comments/comment-navigation/index.jsx (hunk 1 above)
Comment (Copons) [inline client/my-sites/comments/comment-navigation/index.jsx]: Unfortunately `status` is not a prop, but a property of `navItems` (which is generated on the fly by the component itself).

File: client/my-sites/comments/comment-list/index.jsx [hunk 3]
```diff
@@ -189,6 +203,7 @@ export class CommentList extends Component {
 	setCommentStatus = ( comment, status, options = { isUndo: false, doPersist: false, showNotice: true } ) => {
 		const { commentId, postId, isLiked } = comment;
 		const { isUndo, doPersist, showNotice } = options;
+		const alsoUnlike = isLiked && ( 'approved' !== status );
```
Comment (gwwar) [inline client/my-sites/comments/comment-list/index.jsx]: The name reads a bit funny here. How about `unlikeComment`.

File: client/my-sites/comments/comment-list/index.jsx [hunk 4]
```diff
@@ -442,15 +459,57 @@ const mapStateToProps = ( state, { siteId, status } ) => {
 };
 
 const mapDispatchToProps = ( dispatch, { siteId } ) => ( {
-	changeCommentStatus: ( commentId, postId, status ) => dispatch( changeCommentStatus( siteId, postId, commentId, status ) ),
+	changeCommentStatus: ( commentId, postId, status, analytics = { alsoUnlike: false, isUndo: false } ) => dispatch( withAnalytics(
```
Comment (gwwar) [inline client/my-sites/comments/comment-list/index.jsx]: I believe analytics props should be `snake_case`. eg `also_unlike`. I'm not sure if we record correctly other

Comment (gwwar) [inline client/my-sites/comments/comment-list/index.jsx]: If possible I think logging the `last_status` would be useful

File: client/my-sites/comments/comment-list/index.jsx (hunk 3 above)
Comment (Copons) [inline client/my-sites/comments/comment-list/index.jsx]: I'd still keep the also and make it `alsoUnlikeComment` (and then `also_unlike_comment` in the stats payload). We've used this `also` naming on many other places for side effects.

**Pull request merged**